src/cpp/src/continuous_batching_impl.cpp

```diff
@@ -314,27 +350,23 @@ ContinuousBatchingPipeline::ContinuousBatchingImpl::generate(const std::vector<o
             }
         } catch (...) {
             drop_requests(); // remove all requests from pipeline state in case of exception
+            has_active_request = false;
+            cv.notify_all();
             throw;
         }
+        has_active_request = has_non_finished_requests();
+        cv.notify_all();
```

Comment:
```suggestion
        cv.notify_one();
```
as we have one waiting thread